nanoFramework.WebServer/HttpMultipartParser/LineBuffer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+
+namespace nanoFramework.WebServer.HttpMultipartParser
+{
+    internal class LineBuffer : IDisposable
+    {
+        private readonly Hashtable _data = new();
```

Comment:
Why not ArrayList instead?
So, you can just put directly the buffers in there without allocating intermediate buffers to copy the data.

Reply:
No sound reason to use Hashtable, chunk of code copied over from other project where it did make sense. Changed to ArrayList.
The copying of the parts of the buffer is still needed though. The buffer is reused to read chunks of data from the stream.  This is the mechanic behind it:
- Read available bytes from the stream up to buffer.Length
- Search for a newline character. If found: 
 1. Return that part as text if reading a header or a parameter
 2. Return that part as byte[] if reading a file to investigate the read line: the \n could be part of the file or we just read the line that contains the boundary and hence denotes the end of the file.
- On next read of a line we look for newline again in the same buffer until no longer found
- Then we store what's left in the buffer in the linebuffer as the next newline will be in a chunk of data net yet read
- Rinse and repeat: the buffer is reused and filled with the next chunk of data (hence need to copy the remainder of the buffer from previous chunk to the linebuffer)